### PR TITLE
#9490: Added ttnn support for unary ops py file

### PR DESCRIPTION
**Pull request by ruthreshx**
### Ticket
[[9490]](https://github.com/tenstorrent/tt-metal/issues/9490)

### Problem description
Added the ttnn support for unary ops

### What's changed
Replace tt_lib op usage in Python with TTNN

### Checklist
- [x] Post commit CI passes - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9910705610)
- [ ] Nightly fast dispatch same as main - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904592590)
- [x] [post-commit] models tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904608068)
- [x] New/Existing tests provide coverage for changes
- [ ] Device perf regressions and output report - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904626776) 
- [x] Model perf regressions and output report - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904637662)
- [x] T3K unit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904670098)
- [ ] T3K frequent tests - same as main [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904649387)
- [x] T3K demo tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/9904654326)

### CODE REVIEW

**Review by esmalTT — APPROVED** [review]

**Review by boris-drazic — APPROVED** [review]

**Review by mtairum — APPROVED** [review]
Mistral and Mixtral look good 👍 
Just make sure to run t3k demo and t3k unit test pipelines to double check it's passing.

**Review by TT-BrianLiu — APPROVED** [review]

**Review by nemanjagrujic — APPROVED** [review]

**Comment by ruthreshx** [comment]
> Mistral and Mixtral look good 👍 Just make sure to run t3k demo and t3k unit test pipelines to double check it's passing.

Sure will check @mtairum :)

**Review by cglagovichTT — APPROVED** [review]
Llama changes look good

**Review by yieldthought — APPROVED** [review]

**Review by kpaigwar — APPROVED** [review]

**Review by djordje-tt — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]

File: models/experimental/bloom/tt/bloom_gelu_forward.py
Comment (KalaivaniMCW) [inline models/experimental/bloom/tt/bloom_gelu_forward.py:1]: memory_config=mem_config

File: models/experimental/mistral/tt/mistral_attention.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def forward(
 
         scores = ttnn.matmul(query, key, memory_config=self.args.out_mem_config)
         key.deallocate()
-        scores = tt_lib.tensor.mul_unary(scores, self.scale, output_mem_config=self.args.out_mem_config)
+        scores = ttnn.multiply(scores, self.scale, output_mem_config=self.args.out_mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/mistral/tt/mistral_attention.py:203]: memory_config=mem_config

File: models/experimental/roberta/tt/roberta_classification_head.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def forward(self, features, **kwargs):
         x = torch2tt_tensor(torch_x, self.device)
         # x = self.dropout(x)
         x = self.dense_linear(x)
-        x = tt_lib.tensor.tanh(x, output_mem_config=self.mem_config)
+        x = ttnn.tanh(x, output_mem_config=self.mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/roberta/tt/roberta_classification_head.py:69]: memory_config=

File: models/experimental/roberta/tt/roberta_intermediate.py [hunk 3]
```diff
@@ -55,5 +55,5 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
             torch_hidden_states = torch.nn.functional.gelu(torch_hidden_states)
             hidden_states = torch2tt_tensor(torch_hidden_states, self.device)
         else:
-            hidden_states = tt_lib.tensor.gelu(hidden_states, output_mem_config=self.mem_config)
+            hidden_states = ttnn.gelu(hidden_states, output_mem_config=self.mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/roberta/tt/roberta_intermediate.py:58]: memory_config=

File: models/experimental/roberta/tt/roberta_pooler.py [hunk 4]
```diff
@@ -63,5 +63,5 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
         tt_first_token_tensor = torch2tt_tensor(first_token_tensor, self.device)
 
         pooled_output = self.dense_linear(tt_first_token_tensor)
-        pooled_output = tt_lib.tensor.tanh(pooled_output, self.mem_config)
+        pooled_output = ttnn.tanh(pooled_output, self.mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/roberta/tt/roberta_pooler.py:66]: memory_config=

File: models/experimental/vgg/tt/vgg.py [hunk 5]
```diff
@@ -74,15 +75,15 @@ def __init__(
...
-            tt_lib.tensor.relu,
+            ttnn.relu,
             linear3,
         ]
 
     def forward(self, tt_x: tt_lib.tensor.Tensor) -> tt_lib.tensor.Tensor:
         for layer in self.features:
-            if layer is tt_lib.tensor.relu:
+            if layer is ttnn.relu:
                 tt_x = layer(tt_x, output_mem_config=self.output_mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/vgg/tt/vgg.py:87]: memory_config=

File: models/experimental/vit/tt/activations.py [hunk 6]
```diff
@@ -19,15 +19,17 @@
...
-        self.act = tt_lib.tensor.gelu
-        self.out_mem_config_l1 = tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.L1)
-
+        self.act = ttnn.gelu
+        self.out_mem_config_l1 = tt_lib.tensor.MemoryConfig(
+            tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.L1
+        )
 
     def forward(self, input: tt_lib.tensor.Tensor) -> tt_lib.tensor.Tensor:
         return self.act(input, output_mem_config=self.out_mem_config_l1)
```
Comment (KalaivaniMCW) [inline models/experimental/vit/tt/activations.py:35]: memory_config=

File: tests/tt_eager/python_api_testing/fused_ops/silu.py [hunk 7]
```diff
@@ -7,6 +7,6 @@
 
 
 def SiLU(x):
-    xs = ttl.tensor.sigmoid(x)
+    xs = ttnn.sigmoid(x)
     xs = ttnn.mul(xs, x)
```
Comment (KalaivaniMCW) [inline tests/tt_eager/python_api_testing/fused_ops/silu.py:11]: ttnn.multiply

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_unary_ops_ttnn.py [hunk 8]
```diff
@@ -47,3 +47,977 @@ def test_unary_pow_ttnn(input_shapes, exponent, device):
...
+
+    cq_id = 0
+    ttnn.sqrt(input_tensor, output_tensor=output_tensor, queue_id=cq_id)
+    golden_tensor = torch.sqrt(in_data)
+
+    comp_pass = compare_pcc([output_tensor], [golden_tensor])
+    assert comp_pass
+
+
+# ====================================== TYPE ERROR ============================================
```
Comment (KalaivaniMCW) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_unary_ops_ttnn.py:1002]: remove

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_unary_ops_ttnn.py [hunk 9]
```diff
@@ -47,3 +47,977 @@ def test_unary_pow_ttnn(input_shapes, exponent, device):
...
+
+    cq_id = 0
+    ttnn.reciprocal(input_tensor, output_tensor=output_tensor, queue_id=cq_id)
+    golden_tensor = torch.reciprocal(in_data)
+
+    comp_pass = compare_pcc([output_tensor], [golden_tensor])
+    assert comp_pass
+
+
+# ================ Type Error ================
```
Comment (KalaivaniMCW) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_unary_ops_ttnn.py:898]: remove

File: models/experimental/bloom/tt/bloom_gelu_forward.py [hunk 10]
```diff
@@ -49,7 +49,7 @@ def tt_bloom_gelu_forward(x, device):
     factor3 = ttnn.add(factor3, z, memory_config=mem_config)
 
     sumtanh = ttnn.mul(tt_k3, factor3, memory_config=mem_config)
-    tanh = tt_lib.tensor.tanh(sumtanh, output_mem_config=mem_config)
+    tanh = ttnn.tanh(sumtanh, output_mem_config=mem_config)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom/tt/bloom_gelu_forward.py:52]: memory_config=mem_config

**Review by johanna-rock-tt — APPROVED** [review]
falcon40b changes look good

**Review by eyonland — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**